src/smk/tool/geomark/tool-geomark.js

```diff
@@ -286,33 +297,38 @@ include.module( 'tool-geomark', [
 
             smk.on( this.id, {
                 'clear-drawing': function() {
-                    currentLayerGroup.clearLayers();
+                    currentLayerGroupInfo.layerGroup.clearLayers();
                     smk.$viewer.map.pm.disableDraw();
                     smk.$viewer.map.pm.enableDraw();
                     self.canSave = false;
                     self.canClear = false;
                 },
                 'create-geomark': function () {
-                    if (currentLayerGroup.getLayers().length == 0) {
+                    if (currentLayerGroupInfo.layerGroup.getLayers().length == 0) {
                         self.showStatusMessage('No drawings were found. Draw one or more polygons before creating a geomark.', 'warning', 5000);
                         return;
                     }
-                    const wktCoords = self.buildWktCoords(currentLayerGroup);
+                    const wktCoords = self.buildWktCoords(currentLayerGroupInfo.layerGroup);
                     client.createGeomark({
                         'body': 'SRID=4326;' + wktCoords,
                         'format': 'wkt',
-                        'callback': function(geomarkInfo) {
-                            const geomark = self.toGeomark(geomarkInfo, currentLayerGroup);
-                            if (geomark) { 
-                                self.updateAndShowAlert('Created geomark: <a href="' + geomarkInfo.url + 
-                                '" target="_new">' + geomarkInfo.url +
-                                '</a>. Save this URL to access your geomark later.');
-                                self.geomarks.push(geomark);
-                                currentLayerGroup = self.createCurrentLayerGroup();
+                        'callback': function(geomarkJson) {
+                            const geomarkInfo = self.toGeomarkInfo(geomarkJson, currentLayerGroupInfo.layerUUID);
+                            if (geomarkInfo) { 
+                                // Layer group must be removed from map before we can reassign its pane, then added back
+                                smk.$viewer.map.removeLayer(currentLayerGroupInfo.layerGroup);
```

Comment:
Why do you have to reassign the pane with the same value?

Reply:
When creating a geomark, we start by adding polygon layer(s) to the layer group. It seems like the svg element should then get added as a child of the pane's div element, but this doesn't happen, and so without this remove/set-pane/add, the svg stays outside the pane's div. Only by doing all 3 does it seem to be moved (re-added) as a child of the pane div in the DOM. Some more context from a comment in the task I'm tracking this with:

"I ran into a problem with the strategy (of using custom panes). I found this works well when loading a geomark because we create a layer for the polygon(s) and set the custom pane. However this didn’t work as well when creating a geomark, because we create it layer by layer (in case we want a multipolygon) and then add each layer to a FeatureGroup. Unfortunately, Leaflet API favours assigning a layer to a Pane when the layer is created, so it can be added to the DOM without having to worry about moving it later. So a layer effectively cannot be moved from one pane to a different pane.

However I found in [How to change polygon pane in Leaflet (custom pane)](https://stackoverflow.com/questions/42254868/how-to-change-polygon-pane-in-leaflet-custom-pane) that it is possible to remove the layer from the map, change the pane in a setStyle() method, and add it back to the map. This strategy is working for creating a geomark, and so far, I haven’t found any faults after thorough manual testing."